firefox_ui_tests/remote/security/test_mixed_script_content_blocking.py

```diff
@@ -0,0 +1,100 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import By, Wait
+
+from firefox_ui_harness.decorators import skip_under_xvfb
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestMixedScriptContentBlocking(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+        self.test_url = "https://mozqa.com/data/firefox/security/mixed_content_blocked/index.html"
+        self.test_elements = [
+            ('result1', 'Insecure script one'),
+            ('result2', 'Insecure script from iFrame'),
+            ('result3', 'Insecure plugin'),
+            ('result4', 'Insecure stylesheet'),
+        ]
+        self.locationbar = self.browser.navbar.locationbar
+
+    def tearDown(self):
+        self.marionette.set_context("chrome")
+        self.marionette.execute_script("arguments[0].hidePopup();",
+                                       script_args=[self.locationbar.notification_popup])
+        FirefoxTestCase.tearDown(self)
+
+    def _expect_protection_status(self, enabled):
+        color, parts, state = (
+            "rgb(255, 0, 0)",
+            "identity-icons-https-mixed-active.png",
+            "unblocked"
+        )
+        if enabled:
+            color, parts, state = (
+                "rgb(0, 136, 0)",
+                "identity-icons-https.png",
+                "blocked"
+            )
+
+        icon_id = "bad-content-%s-notification-icon" % state
+        icon = self.marionette.find_element(By.ID, icon_id)
+        self.wait_for_condition(lambda _: icon.get_attribute('showing') == 'true')
+        favicon = self.locationbar.favicon
+
+        self.wait_for_condition(
+            lambda _: parts in favicon.value_of_css_property('list-style-image')
+        )
+
+        with self.marionette.using_context('content'):
+            for identifier, description in self.test_elements:
+                el = self.marionette.find_element(By.ID, identifier)
+                Wait(self.marionette).until(
+                    lambda mn: el.value_of_css_property('color') == color,
+                    message=("%s has been %s" % (description, state))
+                )
+
+    def expect_protection_enabled(self):
+        self._expect_protection_status(True)
+
+    def expect_protection_disabled(self):
+        self._expect_protection_status(False)
+
+    @skip_under_xvfb
+    def test_mixed_content_page(self):
+        with self.marionette.using_context('content'):
+            self.marionette.navigate(self.test_url)
+
+        self.expect_protection_enabled()
+
+        panel = self.locationbar.notification_popup
+        self.assertEqual(panel.get_attribute('state'), 'closed')
```

Comment:
Oh, right. So that means that the identity popup is also of the same type, and we would have to convert the class for doorhanger notifications. Mind filing the bug and adding the TODO? We shouldn't forget about it.